CMakeLists.txt

```diff
@@ -57,6 +57,16 @@ set(THEROCK_ARTIFACT_ARCHIVE_SUFFIX "" CACHE STRING "Suffix to add to artifact a
 
 option(THEROCK_BUNDLE_SYSDEPS "Builds bundled system deps for portable builds into lib/rocm_sysdeps" ON)
 
+# Source settings.
+option(THEROCK_USE_EXTERNAL_ROCM_LIBRARIES "Use the `rocm-libraries` monorepo instead of submodules" OFF)
+if(THEROCK_USE_EXTERNAL_ROCM_LIBRARIES AND NOT ROCM_LIBRARIES_SOURCE_DIR)
+  message(FATAL_ERROR "If THEROCK_USE_EXTERNAL_ROCM_LIBRARIES is set, ROCM_LIBRARIES_SOURCE_DIR is required!")
+endif()
```

Comment:
* Should we default to `"${CMAKE_CURRENT_SOURCE_DIR}/../amdgpu-windows-interop"` as is done below? https://github.com/ROCm/TheRock/blob/b49db515a613ee5f98450099cdbe51039704d9a3/CMakeLists.txt#L63
* How about naming `THEROCK_ROCM_LIBRARIES_SOURCE_DIR` for consistency? If subprojects need the unqualified variable name (no `THEROCK_` prefix) for some reason, we could inject it.

Reply:
* For no I would avoid setting a default as it depends on how this will be used in the workflows. Let's therefore not set a default before we have a per-PR testing workflow. I guess we finally move over to consume sources always from the monorepo at which we will add it as a submodule to TheRock. But that's for later ..
* With regards to `THEROCK_ROCM_LIBRARIES_SOURCE_DIR`, that's what I had locally first but wasn't sure. Happy to bring it back and add it to the branch.